packages/portal/src/app/shared/listing.service.ts

```diff
@@ -39,16 +39,27 @@ export class ListingService {
   }
 
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  async share(listing: Listing) {
+  async share(platform: string, listing: Listing) {
     const rent = listing.fees.at(0);
     const currency = listing.fees.at(-1);
-    this.windowService
+    if(platform === "facebook"){
```

Comment:
If we wanted to conform with conventions, there should be a space between the `if` clause and the statement parenthesis. 
We may want here to test if platform is defined at all 

`if (platform && platform === `

But I think we could consider assigning the url value to a conditional value called for example shareUrl. The way I would do it is I would create a global object that has the twitter and facebook urls as properties (so you can extend to more should you need) . 

```
const socialMediaUrls = {
    twitter = '[url-value]',
    facebook = '[url-value]
 ;
```

Then you can import and deconstruct, and execute the conditional for the url value only. // (Sort of quick pseudo code warning here, but you'll get the idea ;) )
```
const { twitter, facebook } = socialMediaUrls;
const shareUrl = (platform) => { platform === 'twitter' ? twitter : facebook; }
```

Then we'd only need to pass shareUrl to the same `this.windowService.native().open(` function.
I see you're also passing the url further down the line and the param name is &u= in one case and &url= in the other. We could also assign that segment of the url to a variable and do the same kind of conditional test to determine what's the corresponding value.

Reply:
@anfibiacreativa Thank you for your amazing feedback I learned a lot!